kernel/mm.c

```diff
@@ -0,0 +1,24 @@
+#include <stdint.h>
+#include "mm.h"
+
+void setup_identity_paging(void)
+{
+    uint32_t i;
+
+    /* set first pte's to first directory */
+    pdir[0] = ((uint32_t) pt) | 0x3;  /* present | r/w | supervisor bits */
+
+    /* fill DIR_SIZE page tables */
+    for(i = 0; i < DIR_SIZE; i++)
+        pt[i] = (i * FRAME) | 0x3; /* present | supervisor bits */
```

Comment:
I don't think this will work like this becasue you are not explicitly setting the physical address of the page, so it may start at 0x0 or whatever uninitialized values live in that part of the struct.

Reply:
Nevermind (i*FRAME) is the phisycal address , my bad :(